packages/shared/src/components/search/SearchBarInput.tsx

```diff
@@ -81,43 +92,67 @@ function SearchBarInputComponent(
     setInput('');
   };
 
-  const onSubmit = (event: MouseEvent): void => {
-    event.stopPropagation();
+  const onSubmit = (event: FormEvent, input?: string): void => {
+    event.preventDefault();
+
+    if (!user) return showLogin('search input');
+
+    const finalValue = input ?? inputRef.current.value;
 
     if (handleSubmit) {
-      handleSubmit(event, inputRef.current.value);
+      handleSubmit(event, finalValue);
     }
 
-    setInput(null);
+    return setInput(finalValue);
   };
 
   const seeSearchHistory = (event: MouseEvent): void => {
     event.stopPropagation();
   };
 
+  const onInputClick = () => {
+    if (isTabletAbove || isMobileOpen || !shouldShowPopup) return null;
+
+    if (!user) return showLogin('search input');
```

Comment:
This should be a value of AuthTriggers, I don't know who keeps adding the string expection, but it's not expected 😅

Reply:
Oh, right. That is correct.  Let me update this one. Probably due to some forgotten TODOs.